base/special/trig.jl

```diff
@@ -76,6 +76,105 @@ function cos_kernel(x::DoubleFloat32)
     Float32(((1.0+z*C0) + w*C1) + (w*z)*r)
 end
 
+# asin methods
+ASIN_X_MIN_THRESHOLD(::Type{Float32}) = 2.0f0^-12
+ASIN_X_MIN_THRESHOLD(::Type{Float64}) = sqrt(eps(Float64))
+
+asin_p(t::Float64) = t*@horner(t,
+                             1.66666666666666657415e-01,
+                            -3.25565818622400915405e-01,
+                             2.01212532134862925881e-01,
+                            -4.00555345006794114027e-02,
+                             7.91534994289814532176e-04,
+                             3.47933107596021167570e-05)
+
+asin_q(t::Float64) = @horner(t,
+                             1.0,
+                            -2.40339491173441421878e+00,
+                             2.02094576023350569471e+00,
+                            -6.88283971605453293030e-01,
+                             7.70381505559019352791e-02)
+
+asin_p(t::Float32) = t*@horner(t,
+                             Float32(1.6666586697e-01),
+                             Float32(-4.2743422091e-02),
+                             Float32(-8.6563630030e-03))
+
+asin_q(t::Float32) = 1.0f0+t*Float32(-7.0662963390e-01)
```

Comment:
Haven't had a change to look at things in detail but just use the float syntax here and above, e.g.
`-7.0662963390f-01`

seems like this could also benefit from `muladd`